src/libmongoc/CMakeLists.txt

```diff
@@ -810,8 +835,11 @@ if (MONGOC_ENABLE_STATIC_BUILD)
          $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/../../src>
          $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/../../src/common>
    )
-   set_target_properties (mongoc_static PROPERTIES VERSION 0.0.0)
-   set_target_properties (mongoc_static PROPERTIES OUTPUT_NAME "${MONGOC_OUTPUT_BASENAME}-static-${MONGOC_API_VERSION}")
+   set_target_properties (mongoc_static PROPERTIES
+      VERSION 0.0.0
+      OUTPUT_NAME "${MONGOC_OUTPUT_BASENAME}-static-${MONGOC_API_VERSION}"
+      pc_REQUIRES "libbson-static-1.0"
```

Comment:
I can rename it to `pkgconfig_` or `pkg_config_`, but I'll still keep it lowercase, because that's the under-documented way to "correctly" name custom properties. CMake arcana:

CMake won't let you use arbitrary properties on `INTERFACE` libraries. CMake will yell at you if you even _look for_ a property that isn't "allowed" on an INTERFACE library. I think this restriction is obnoxious and pointless, and even if they remove(d?) it then there are still CMake executables out there that will yell at you.

Of the properties permitted: anything that starts with an underscore or a lowercase letter. I'm still scratching my head about it. Oh well.

Reply:
Perhaps the property must be prefixed with `INTERFACE_`? From https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#compatible-interface-properties:
> Some target properties are required to be compatible between a target and the interface of each dependency. For example, the `POSITION_INDEPENDENT_CODE` target property may specify a boolean value [...] To be "compatible", the `POSITION_INDEPENDENT_CODE` property, if set must be either the same, in a boolean sense, as the `INTERFACE_POSITION_INDEPENDENT_CODE` property of all transitively specified dependencies on which that property is set. This property of "compatible interface requirement" may be extended to other properties by specifying the property in the content of the `COMPATIBLE_INTERFACE_BOOL` target property. Each specified property must be compatible between the consuming target and the corresponding property with an `INTERFACE_` prefix from each dependency: [...]

Reply:
This is confusing (I know it comes from the documentation):

> if set must be either the same, in a boolean sense, as the INTERFACE_POSITION_INDEPENDENT_CODE property of all transitively specified dependencies on which that property is set

They used "either" but there is no corresponding "or". I wonder if this is a result of a change where an "or" had been present, but removed, and the structure of the rest of the sentence wasn't updated.

Reply:
The `INTERFACE_` target properties are "whitelisted" for `INTERFACE` libraries, as well as the `COMPATIBLE_INTERFACE_` properties, but those are to deal with transitive usage requirements when linking targets together, not per-target custom properties.